apps/api/src/app/events/usecases/send-message/send-message-push.usecase.ts

```diff
@@ -0,0 +1,196 @@
+    const subscriber: SubscriberEntity = await this.subscriberRepository.findOne({
+      _environmentId: command.environmentId,
+      _id: command.subscriberId,
+    });
+    const contentService = new ContentService();
+    const messageVariables = contentService.buildMessageVariables(command.payload, subscriber);
+    const content = contentService.replaceVariables(pushChannel.template.content as string, messageVariables);
+    const title = contentService.replaceVariables(pushChannel.template.title as string, messageVariables);
+    const notificationIdentifiers = command.payload.notificationIdentifiers || subscriber.notificationIdentifiers;
+
+    const messagePayload = Object.assign({}, command.payload);
+    delete messagePayload.attachments;
+
+    const message: MessageEntity = await this.messageRepository.create({
+      _notificationId: notification._id,
+      _environmentId: command.environmentId,
+      _organizationId: command.organizationId,
+      _subscriberId: command.subscriberId,
+      _templateId: notification._templateId,
+      _messageTemplateId: pushChannel.template._id,
+      channel: ChannelTypeEnum.PUSH,
+      transactionId: command.transactionId,
+      notificationIdentifiers,
+      content,
+      title,
+      payload: messagePayload,
+    });
+
+    const integration = await this.integrationRepository.findOne({
+      _environmentId: command.environmentId,
+      channel: ChannelTypeEnum.PUSH,
+      active: true,
+    });
+
+    if (notificationIdentifiers && integration) {
+      await this.sendMessage(
+        integration,
+        notificationIdentifiers,
+        title,
+        content,
+        message,
+        command,
+        notification,
+        command.payload
+      );
+
+      return;
+    }
+
+    await this.sendErrors(notificationIdentifiers, integration, message, command, notification);
+  }
+
+  private async sendErrors(
+    phone,
+    integration,
+    message: MessageEntity,
+    command: SendMessageCommand,
+    notification: NotificationEntity
+  ) {
+    if (!phone) {
+      await this.createLogUsecase.execute(
+        CreateLogCommand.create({
+          transactionId: command.transactionId,
+          status: LogStatusEnum.ERROR,
+          environmentId: command.environmentId,
+          organizationId: command.organizationId,
+          text: 'Subscriber does not have active phone',
+          userId: command.userId,
+          subscriberId: command.subscriberId,
+          code: LogCodeEnum.SUBSCRIBER_MISSING_PHONE,
+          templateId: notification._templateId,
+          raw: {
+            payload: command.payload,
+            triggerIdentifier: command.identifier,
+          },
+        })
+      );
+      await this.messageRepository.updateMessageStatus(
+        message._id,
+        'warning',
+        null,
+        'no_subscriber_phone',
+        'Subscriber does not have active phone'
+      );
```

Comment:
This one is not really a `phone` property, maybe add an enum for missing push credentials? So we can separate them from and actual phone for SMS. Wdyt?

Reply:
Was left out on accident, fixed.